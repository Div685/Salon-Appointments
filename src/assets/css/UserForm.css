.form__section .form {
  display: flex;
  flex-direction: column;
}