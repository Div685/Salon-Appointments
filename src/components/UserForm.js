import React, { useState } from 'react';
import PropTypes from 'prop-types';

const UserForm = ({ handleSubmit, btnName }) => {
  const [userName, setUserName] = useState('');
  const [password, setPassword] = useState('');

  const onSubmitForm = (e) => {
    e.preventDefault();
    handleSubmit(userName, password);
  };

  return (
    <div className="form__section">
      <form className="form" onSubmit={onSubmitForm}>
        <input
          className="form__input"
          placeholder="Username"
          type="text"
          name="username"
          value={userName}
          onChange={(event) => { setUserName(event.target.value); }}
        />
        <input
          className="form__input"
          placeholder="password"
          type="password"
          name="password"
          value={password}
          onChange={(event) => { setPassword(event.target.value); }}
        />

        <button type="submit" className="btn__login">{btnName}</button>
      </form>
    </div>
  );
};

UserForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  btnName: PropTypes.string.isRequired,
};

export default UserForm;
