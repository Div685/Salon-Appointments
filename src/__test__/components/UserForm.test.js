import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import UserForm from '../../components/UserForm';
import configureStore from '../helper/createStore';

describe('UsersForm', () => {
  let store;
  beforeEach(() => {
    store = configureStore();
  });
  test('should match with snapshot', () => {
    const handleSubmit = jest.fn();
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <UserForm handleSubmit={handleSubmit} btnName="Signup" />
          </MemoryRouter>
        </Provider>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
