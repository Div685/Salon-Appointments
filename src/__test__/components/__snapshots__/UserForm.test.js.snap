// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersForm should match with snapshot 1`] = `
<div
  className="form__section"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <input
      className="form__input"
      name="username"
      onChange={[Function]}
      placeholder="Username"
      type="text"
      value=""
    />
    <input
      className="form__input"
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value=""
    />
    <button
      className="btn__login"
      type="submit"
    >
      Signup
    </button>
  </form>
</div>
`;
